### ISSUE TITLE
Make twisted.enterprise Python 3 ready

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8303]
|[<img alt="oberstet's avatar" src="https://avatars.githubusercontent.com/u/233340?s=50" width="50" height="50">](https://github.com/oberstet)| @oberstet reported|
|-|-|
|Trac ID|trac#8303|
|Type|enhancement|
|Created|2016-04-25 17:44:47Z|
|Branch|https://github.com/twisted/twisted/tree/8303-rodrigc-enterprise-py3|

`twisted.enterprise` seems not be ported to Python 3.

There is no warning, but a hard import error:

```
(cpy351_1) oberstet@bvr-sql18:~$ python
Python 3.5.1 (default, Apr 25 2016, 17:57:33)
[GCC 4.9.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from twisted.enterprise import adbapi
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'twisted.enterprise'
>>> import twisted.enterprise
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'twisted.enterprise'
>>>

```

There is also no mention about on the Python 3 porting progress page here [https://twistedmatrix.com/trac/wiki/Plan/Python3]

<details><summary>Searchable metadata</summary>

```
trac-id__8303 8303
type__enhancement enhancement
reporter__oberstet oberstet
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8303_rodrigc_enterprise_py3 8303-rodrigc-enterprise-py3
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1461606287354545 1461606287354545
changetime__1467199376402508 1467199376402508
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [47319]) Branching to adbapi-py3-8303.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

I submitted one patch for this:

https://github.com/twisted/twisted/pull/74

Please review

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @rodrigc|
|-|-|

If this is about porting twisted.enterprise to py3, then we also need to enable the twisted.enterprise  tests on py3.

If twisted.enterprise  is now ported on py3, then the branch should have a news file fragment, advertising this new feature.

The description inside the PR looks like a generic py3 porting update and not something targeted for having twisted.enterprise ported on py3.

Anyway we need to new changes executed on Py3 so that we can make sure that the changes are ok :)

Please review my comments.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

Amber has already start working on this.

Please see her branch and diff here https://github.com/twisted/twisted/compare/trunk...adbapi-py3-8303

it would be nice to continue based on her work

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set owner to @hawkowl|
|-|-|

Can you review this?  I provided feedback in your branch: https://github.com/twisted/twisted/compare/trunk...adbapi-py3-8303

The changes you made in your branch to change raising exceptions, and catching exceptions can go away because they are already in trunk.

It will be good to combine efforts on this patch.

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

I am deeply confused about what's up for review here, and by whom.  There appear to be changes from hawkowl, commented upon by rodrigc, but ... re-assigned to hawkowl, but in review?  Does this need a code review, and if so, does it need to be from hawkowl?

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc commented|
|-|-|

Glyph, I understand the confusion.  There are multiple efforts going on in parallel with this ticket.  I am going to take this ticket out of review, and
submit my changes (which are a smaller subset than what Amber has worked on) in a separate ticket for review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_11"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/240

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_12"></a>|@Lukasa set owner to @rodrigc|
|-|-|

LGTM, merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_13"></a>|GitHub set status to `closed`|
|-|-|

In changeset 68f04e7373705ea6356991259e34d83fc00bf81b
```
#!CommitTicketReference repository="" revision="68f04e7373705ea6356991259e34d83fc00bf81b"
Merge 8303-rodrigc-enterprise-py3: Add twisted.enterprise.adbapi to list of Python 3 ported modules

Author: rodrigc
Reviewer: Lukasa
Fixes: #8303
```
